File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/DflComputeStatusResult.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Status of DI Agent enablement and managed compute lifecycle for an AI Data Platform instance.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=DflComputeStatusResult.Builder.class)

public final class DflComputeStatusResult  {
    @Deprecated
    @java.beans.ConstructorProperties({"isEnabled", "lifecycleState", "enablementStatus", "statusDetails"})
    public DflComputeStatusResult(Boolean isEnabled, Cluster.State lifecycleState, String enablementStatus, String statusDetails) {
        super();
        this.isEnabled = isEnabled;
        this.lifecycleState = lifecycleState;
        this.enablementStatus = enablementStatus;
        this.statusDetails = statusDetails;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Whether DI Agent enablement is enabled for the AI Data Platform instance.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("isEnabled")
private Boolean isEnabled;

        /**
         * Whether DI Agent enablement is enabled for the AI Data Platform instance.
         * @param isEnabled the value to set
         * @return this builder
         **/
        

public Builder isEnabled(Boolean isEnabled) {
    this.isEnabled = isEnabled;
    return this;
}
            /**
     * The current DI Agent managed compute lifecycle state when compute exists.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("lifecycleState")
private Cluster.State lifecycleState;

        /**
         * The current DI Agent managed compute lifecycle state when compute exists.
         * @param lifecycleState the value to set
         * @return this builder
         **/
        

public Builder lifecycleState(Cluster.State lifecycleState) {
    this.lifecycleState = lifecycleState;
    return this;
}
            /**
     * Durable DI Agent enablement state, such as InProgress, Succeeded, or Failed.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("enablementStatus")
private String enablementStatus;

        /**
         * Durable DI Agent enablement state, such as InProgress, Succeeded, or Failed.
         * @param enablementStatus the value to set
         * @return this builder
         **/
        

public Builder enablementStatus(String enablementStatus) {
    this.enablementStatus = enablementStatus;
    return this;
}
            /**
     * Additional sanitized details for the durable enablement state, populated when available.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("statusDetails")
private String statusDetails;

        /**
         * Additional sanitized details for the durable enablement state, populated when available.
         * @param statusDetails the value to set
         * @return this builder
         **/
        

public Builder statusDetails(String statusDetails) {
    this.statusDetails = statusDetails;
    return this;
}


        public DflComputeStatusResult build() {
            DflComputeStatusResult model = new DflComputeStatusResult(this.isEnabled
                , this.lifecycleState
                , this.enablementStatus
                , this.statusDetails);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(DflComputeStatusResult model) {
                this.isEnabled(model.getIsEnabled());
    this.lifecycleState(model.getLifecycleState());
    this.enablementStatus(model.getEnablementStatus());
    this.statusDetails(model.getStatusDetails());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Whether DI Agent enablement is enabled for the AI Data Platform instance.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("isEnabled")
    private final Boolean isEnabled;

        /**
     * Whether DI Agent enablement is enabled for the AI Data Platform instance.
     * @return the value
     **/
    
    public Boolean getIsEnabled() {
        return isEnabled;
    }


        /**
     * The current DI Agent managed compute lifecycle state when compute exists.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("lifecycleState")
    private final Cluster.State lifecycleState;

        /**
     * The current DI Agent managed compute lifecycle state when compute exists.
     * @return the value
     **/
    
    public Cluster.State getLifecycleState() {
        return lifecycleState;
    }


        /**
     * Durable DI Agent enablement state, such as InProgress, Succeeded, or Failed.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("enablementStatus")
    private final String enablementStatus;

        /**
     * Durable DI Agent enablement state, such as InProgress, Succeeded, or Failed.
     * @return the value
     **/
    
    public String getEnablementStatus() {
        return enablementStatus;
    }


        /**
     * Additional sanitized details for the durable enablement state, populated when available.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("statusDetails")
    private final String statusDetails;

        /**
     * Additional sanitized details for the durable enablement state, populated when available.
     * @return the value
     **/
    
    public String getStatusDetails() {
        return statusDetails;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("DflComputeStatusResult(");
        sb.append("isEnabled=").append(String.valueOf(this.isEnabled));
        sb.append(", lifecycleState=").append(String.valueOf(this.lifecycleState));
        sb.append(", enablementStatus=").append(String.valueOf(this.enablementStatus));
        sb.append(", statusDetails=").append(String.valueOf(this.statusDetails));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof DflComputeStatusResult)) {
            return false;
        }

        DflComputeStatusResult other = (DflComputeStatusResult) o;
        return java.util.Objects.equals(this.isEnabled, other.isEnabled) &&
            java.util.Objects.equals(this.lifecycleState, other.lifecycleState) &&
            java.util.Objects.equals(this.enablementStatus, other.enablementStatus) &&
            java.util.Objects.equals(this.statusDetails, other.statusDetails);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.isEnabled == null ? 43 : this.isEnabled.hashCode());
        result = (result * PRIME) + (this.lifecycleState == null ? 43 : this.lifecycleState.hashCode());
        result = (result * PRIME) + (this.enablementStatus == null ? 43 : this.enablementStatus.hashCode());
        result = (result * PRIME) + (this.statusDetails == null ? 43 : this.statusDetails.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/dfl_compute_status_result.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class DflComputeStatusResult(object):
    """
    Status of DI Agent enablement and managed compute lifecycle for an AI Data Platform instance.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new DflComputeStatusResult object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param is_enabled:
            The value to assign to the is_enabled property of this DflComputeStatusResult.
        :type is_enabled: bool

        :param lifecycle_state:
            The value to assign to the lifecycle_state property of this DflComputeStatusResult.
        :type lifecycle_state: str

        :param enablement_status:
            The value to assign to the enablement_status property of this DflComputeStatusResult.
        :type enablement_status: str

        :param status_details:
            The value to assign to the status_details property of this DflComputeStatusResult.
        :type status_details: str

        """
        self.swagger_types = {
            'is_enabled': 'bool',
            'lifecycle_state': 'str',
            'enablement_status': 'str',
            'status_details': 'str'
        }

        self.attribute_map = {
            'is_enabled': 'isEnabled',
            'lifecycle_state': 'lifecycleState',
            'enablement_status': 'enablementStatus',
            'status_details': 'statusDetails'
        }

        self._is_enabled = None
        self._lifecycle_state = None
        self._enablement_status = None
        self._status_details = None

    @property
    def is_enabled(self):
        """
        **[Required]** Gets the is_enabled of this DflComputeStatusResult.
        Whether DI Agent enablement is enabled for the AI Data Platform instance.


        :return: The is_enabled of this DflComputeStatusResult.
        :rtype: bool
        """
        return self._is_enabled

    @is_enabled.setter
    def is_enabled(self, is_enabled):
        """
        Sets the is_enabled of this DflComputeStatusResult.
        Whether DI Agent enablement is enabled for the AI Data Platform instance.


        :param is_enabled: The is_enabled of this DflComputeStatusResult.
        :type: bool
        """
        self._is_enabled = is_enabled

    @property
    def lifecycle_state(self):
        """
        Gets the lifecycle_state of this DflComputeStatusResult.
        The current DI Agent managed compute lifecycle state when compute exists.


        :return: The lifecycle_state of this DflComputeStatusResult.
        :rtype: str
        """
        return self._lifecycle_state

    @lifecycle_state.setter
    def lifecycle_state(self, lifecycle_state):
        """
        Sets the lifecycle_state of this DflComputeStatusResult.
        The current DI Agent managed compute lifecycle state when compute exists.


        :param lifecycle_state: The lifecycle_state of this DflComputeStatusResult.
        :type: str
        """
        self._lifecycle_state = lifecycle_state

    @property
    def enablement_status(self):
        """
        Gets the enablement_status of this DflComputeStatusResult.
        Durable DI Agent enablement state, such as InProgress, Succeeded, or Failed.


        :return: The enablement_status of this DflComputeStatusResult.
        :rtype: str
        """
        return self._enablement_status

    @enablement_status.setter
    def enablement_status(self, enablement_status):
        """
        Sets the enablement_status of this DflComputeStatusResult.
        Durable DI Agent enablement state, such as InProgress, Succeeded, or Failed.


        :param enablement_status: The enablement_status of this DflComputeStatusResult.
        :type: str
        """
        self._enablement_status = enablement_status

    @property
    def status_details(self):
        """
        Gets the status_details of this DflComputeStatusResult.
        Additional sanitized details for the durable enablement state, populated when available.


        :return: The status_details of this DflComputeStatusResult.
        :rtype: str
        """
        return self._status_details

    @status_details.setter
    def status_details(self, status_details):
        """
        Sets the status_details of this DflComputeStatusResult.
        Additional sanitized details for the durable enablement state, populated when available.


        :param status_details: The status_details of this DflComputeStatusResult.
        :type: str
        """
        self._status_details = status_details

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
